documentation/API_documentation/consent-info.yaml

```diff
@@ -0,0 +1,387 @@
+
+    **TBC**
+
+    # Authorization and authentication
+
+    The "Camara Security and Interoperability Profile" provides details of how an API consumer requests an access token. Please refer to Identity and Consent Management (https://github.com/camaraproject/IdentityAndConsentManagement/) for the released version of the profile.
+
+    The specific authorization flows to be used will be agreed upon during the onboarding process, happening between the API consumer and the API provider, taking into account the declared purpose for accessing the API, whilst also being subject to the prevailing legal framework dictated by local legislation.
+
+    In cases where personal data is processed by the API and users can exercise their rights through mechanisms such as opt-in and/or opt-out, the use of three-legged access tokens is mandatory. This ensures that the API remains in compliance with privacy regulations, upholding the principles of transparency and user-centric privacy-by-design.
+    
+    # Further info and support
+
+    (FAQs will be added in a later version of the documentation)
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+  version: wip
+  x-camara-commonalities: 0.5
+servers:
+  - url: '{apiRoot}/consent-info/vwip'
+    variables:
+      apiRoot:
+        default: http://localhost:9091
+        description: API root, defined by the service provider, e.g. `api.example.com` or `api.example.com/somepath`
+tags:
+  - name: Verify Consent Status
+    description: Create a request to verify the Consent status
+paths: 
+  /verify:
+    post:
+      summary: Create a request to verify the Consent status
+      description: |
+        Create a request to verify the Consent status of a specific User to the API Consumer for the requested scope(s) and Purpose. The User is identified by the `phoneNumber` parameter in the request body. The API Consumer is identified by the `client_id` parameter deduced from the access token.
+      operationId: verifyConsentStatus
+      security:
+        - openId:
+          - consent-info:verify
+      tags:
+        - Verify Consent Status      
+      parameters:
+        - $ref: '#/components/parameters/x-correlator'
+      requestBody:
+        required: true
+        description:
+          Consent status verification request
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/VerifyConsentStatusRequestBody"
+            example:
+              phoneNumber: "+123456789"
+              scopes:
+                - "location-verification:verify"
+              purpose: "dpv:FraudPreventionAndDetection"
```

Comment:
Is there a reason why we couldn't allow a list of scope/purposes on this API? It would be nice to be able to have a single call to obtain all of the statuses at once, and/or receive the consent URL to collect consent for all of those scope/purposes?

Reply:
I think this comment can be discussed when we review the actual API spec, rather than whether the API should exist; I am/Vonage is in agreement that this API is required and is much more suitable/scaleable than using Auth Code (prompt=none) for this purpose.